client/state/data-layer/wpcom/sites/comments/likes/new/index.js

```diff
@@ -23,13 +25,29 @@ export const likeComment = ( { dispatch }, action ) => {
 	}, action ) );
 };
 
-export const updateCommentLikes = ( { dispatch }, { siteId, postId, commentId }, next, { like_count } ) => dispatch( local( {
-	type: COMMENTS_LIKE,
-	siteId,
-	postId,
-	commentId,
-	like_count
-} ) );
+export const updateCommentLikes = ( { dispatch }, { siteId, postId, commentId, status }, next, { like_count } ) => {
+	dispatch( local( {
+		type: COMMENTS_LIKE,
+		siteId,
+		postId,
+		commentId,
+		like_count
+	} ) );
+
+	if ( isEnabled( 'comments/undo-in-datalayer' ) && 'unapproved' === status ) {
+		const noticeId = `comment-notice-${ commentId }`;
+
+		dispatch( removeNotice( noticeId ) );
+
+		dispatch( successNotice( translate( 'Comment approved.' ), {
+			duration: 5000,
+			id: noticeId,
+			isPersistent: true,
+			button: translate( 'Undo' ),
+			onClick: () => dispatch( changeCommentStatus( siteId, postId, commentId, 'unapproved' ) )
```

Comment:
The plan is/was to keep the comment around until page refresh or filter change. So, the user would approve a comment, and it would not disappear. If the user changes tabs or reloads the page, then the comment will be in its proper place. This behavior is the same as the Unread filter in Mail.app on iOS and others (like our own notifications).

We just haven't gotten that far in the project yet. Probably will need to be an m2 thing. I actually just noticed that there is not an issue for it, sorry about that. I think we've talked about it in Slack, but just haven't documented it.

I think it's still valuable to have a notice when a user performs an action. Including the Undo button will instill confidence that they can revert an action if they ever do it accidentally. They can of course revert without the Undo button as well.